File: CS/Form1.cs
using System;
using System.Drawing;
using System.Windows.Forms;
using DevExpress.XtraMap;

namespace ChartItemsKeyValueColorizer {
    public partial class Form1 : Form {

        public Form1() {
            InitializeComponent();
            InitializeMap();
        }

        private void InitializeMap() {
            // Create a map control with initial settings and add it to the form.
            MapControl map = new MapControl() { Dock = DockStyle.Fill };
            this.Controls.Add(map);

            //Create a tiles layer and add it to the Map control.
            map.Layers.Add(new ImageTilesLayer() {
                DataProvider = new OpenStreetMapDataProvider()
            });
            
            // Create a vector items layer and add it to the Map control.
            VectorItemsLayer itemsLayer = new VectorItemsLayer() {
                Data = CreateData(),
                Colorizer = CreateColorizer() 
            };
            map.Layers.Add(itemsLayer);

            // Create a legend.
            map.Legends.Add(new ColorListLegend() { Layer = itemsLayer });
        }

        private MapDataAdapterBase CreateData() {
            MapItemStorage storage = new MapItemStorage();
            
            // Create bubble charts and add them to the storage.
            storage.Items.Add(new MapBubble() { Argument = "A", Value = 100, 
                Location = new GeoPoint(-50, -70), Size = 100 
            });
            storage.Items.Add(new MapBubble() { Argument = "B", Value = 70, 
                Location = new GeoPoint(-50, 0), Size = 70 
            });
            storage.Items.Add(new MapBubble() { Argument = "C", Value = 120, 
                Location = new GeoPoint(-50, 70), Size = 120 
            });
            
            //Create a pie chart and add it to the storage.
            MapPie pie = new MapPie() { Location = new GeoPoint(30, 0), Size = 100 };
            pie.Segments.Add(new PieSegment() { Argument = "A", Value = 100 });
            pie.Segments.Add(new PieSegment() { Argument = "B", Value = 50 });
            pie.Segments.Add(new PieSegment() { Argument = "C", Value = 120 });
            storage.Items.Add(pie);

            return storage;
        }

        private MapColorizer CreateColorizer() {
            KeyValueColorizer colorizer = new KeyValueColorizer() {
                ColorItemKeyProvider = new ChartItemArgumentToColorKeyProvider()
            };
            colorizer.AddItem("A", new ColorizerColorTextItem(Color.Coral, "Argument A"));
            colorizer.AddItem("B", new ColorizerColorTextItem(Color.Orange, "Argument B"));
            colorizer.AddItem("C", new ColorizerColorTextItem(Color.LightBlue, "Argument C"));

            return colorizer;
        }
    }

}

File: VB/Form1.vb
Imports System
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraMap

Namespace ChartItemsKeyValueColorizer
    Partial Public Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
            InitializeMap()
        End Sub

        Private Sub InitializeMap()
            ' Create a map control with initial settings and add it to the form.
            Dim map As New MapControl() With {.Dock = DockStyle.Fill}
            Me.Controls.Add(map)

            'Create a tiles layer and add it to the Map control.
            map.Layers.Add(New ImageTilesLayer() With {.DataProvider = New OpenStreetMapDataProvider()})

            ' Create a vector items layer and add it to the Map control.
            Dim itemsLayer As New VectorItemsLayer() With {.Data = CreateData(), .Colorizer = CreateColorizer()}
            map.Layers.Add(itemsLayer)

            ' Create a legend.
            map.Legends.Add(New ColorListLegend() With {.Layer = itemsLayer})
        End Sub

        Private Function CreateData() As MapDataAdapterBase
            Dim storage As New MapItemStorage()

            ' Create bubble charts and add them to the storage.
            storage.Items.Add(New MapBubble() With {.Argument = "A", .Value = 100, .Location = New GeoPoint(-50, -70), .Size = 100})
            storage.Items.Add(New MapBubble() With {.Argument = "B", .Value = 70, .Location = New GeoPoint(-50, 0), .Size = 70})
            storage.Items.Add(New MapBubble() With {.Argument = "C", .Value = 120, .Location = New GeoPoint(-50, 70), .Size = 120})

            'Create a pie chart and add it to the storage.
            Dim pie As New MapPie() With {.Location = New GeoPoint(30, 0), .Size = 100}
            pie.Segments.Add(New PieSegment() With {.Argument = "A", .Value = 100})
            pie.Segments.Add(New PieSegment() With {.Argument = "B", .Value = 50})
            pie.Segments.Add(New PieSegment() With {.Argument = "C", .Value = 120})
            storage.Items.Add(pie)

            Return storage
        End Function

        Private Function CreateColorizer() As MapColorizer
            Dim colorizer As New KeyValueColorizer() With {.ColorItemKeyProvider = New ChartItemArgumentToColorKeyProvider()}
            colorizer.AddItem("A", New ColorizerColorTextItem(Color.Coral, "Argument A"))
            colorizer.AddItem("B", New ColorizerColorTextItem(Color.Orange, "Argument B"))
            colorizer.AddItem("C", New ColorizerColorTextItem(Color.LightBlue, "Argument C"))

            Return colorizer
        End Function
    End Class

End Namespace

